pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -251,379 +150,103 @@ def test_file_decriptors(self):
-            mutated_updaterecords[RPM_UPDATERECORD_ID],
-        )
-
-
-class SyncDiffChecksumPackagesTestCase(unittest.TestCase):
-    """Syncing duplicate NEVRA with different checksums."""
-
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        cls.cfg = config.get_config()
-        cls.client = api.Client(cls.cfg, api.json_handler)
+        task = self.do_test("http://i-am-an-invalid-url.com/invalid/")
+        self.assertIsNotNone(task["error"]["description"])
 
-    def test_all(self):
-        """Sync two fixture content with same NEVRA and different checksum.
+    def test_invalid_rpm_content(self):
+        """Sync a repository using an invalid plugin_content repository.
 
-        Make sure we end up with the most recently synced content.
-
-        Do the following:
-
-        1. Create a repository
-        2. Create two remotes with same content but different checksums.
-            Sync the remotes one after the other.
-               a. Sync remote with packages with SHA256: ``RPM_UNSIGNED_FIXTURE_URL``.
-               b. Sync remote with packages with SHA512: ``RPM_SHA512_FIXTURE_URL``.
-        3. Make sure the latest content is only kept.
-
-        This test targets the following issues:
-
-        * `Pulp #4297 <https://pulp.plan.io/issues/4297>`_
-        * `Pulp #3954 <https://pulp.plan.io/issues/3954>`_
+        Assert that an exception is raised, and that error message has
+        keywords related to the reason of the failure. See :meth:`do_test`.
         """
-        # Step 1
-        repo = self.client.post(RPM_REPO_PATH, gen_repo())
-        self.addCleanup(self.client.delete, repo['pulp_href'])
-
-        # Step 2.
-
-        for body in [
-            gen_rpm_remote(),
-            gen_rpm_remote(url=RPM_SHA512_FIXTURE_URL),
-        ]:
-            remote = self.client.post(RPM_REMOTE_PATH, body)
-            self.addCleanup(self.client.delete, remote['pulp_href'])
-            # Sync the repository.
-            sync(self.cfg, remote, repo)
-
-        # Step 3
-        repo = self.client.get(repo['pulp_href'])
-        added_content = get_content(repo)[RPM_PACKAGE_CONTENT_NAME]
-        removed_content = get_removed_content(repo)[RPM_PACKAGE_CONTENT_NAME]
-
-        # In case of "duplicates" the most recent one is chosen, so the old
-        # package is removed from and the new one is added to a repo version.
-        self.assertEqual(len(added_content), RPM_PACKAGE_COUNT, added_content)
-        self.assertEqual(
-            len(removed_content), RPM_PACKAGE_COUNT, removed_content
-        )
-
-        # Verifying whether the packages with first checksum is removed and second
-        # is added.
-        self.assertEqual(added_content[0]['checksum_type'], 'sha512')
-        self.assertEqual(removed_content[0]['checksum_type'], 'sha256')
-
-
-class ChecksumConstraintTestCase(unittest.TestCase):
```

Comment:
we need to convert this test case also

Reply:
converted as 'test_checksum_constraint' as not need really it is own class